index.js

```diff
@@ -564,9 +564,14 @@ function buildArray (context, location) {
 
   if (Array.isArray(itemsSchema)) {
     for (let i = 0; i < itemsSchema.length; i++) {
-      const item = itemsSchema[i]
+      let item = itemsSchema[i]
+      let itemLocation = itemsLocation.getPropertyLocation(i)
+      if (item.$ref) {
+        itemLocation = resolveRef(context, itemLocation)
+        item = itemLocation.schema
+      }
       functionCode += `value = obj[${i}]`
-      const tmpRes = buildValue(context, itemsLocation.getPropertyLocation(i), 'value')
+      const tmpRes = buildValue(context, itemLocation, 'value')
```

Comment:
Shouldn't `buildValue` itself will resolve `$ref`?

Reply:
Maybe, I only looked into the code enough to fix the problem with minimal changes.  But my problem wasn't with buildValue, it was that buildArrayTypeCondition was given item.type which was undefined since item was still {$ref }.